lint/package.go

```diff
@@ -182,17 +184,22 @@ func (p *Package) scanSortable() {
 	}
 }
 
-func (p *Package) lint(rules []Rule, config Config, failures chan Failure) {
+func (p *Package) lint(rules []Rule, config Config, failures chan Failure) error {
 	p.scanSortable()
 	var wg sync.WaitGroup
 	for _, file := range p.files {
 		wg.Add(1)
 		go (func(file *File) {
-			file.lint(rules, config, failures)
+			err := file.lint(rules, config, failures)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, "Error during linting:", err)
+				os.Exit(1)
+			} 
```

Comment:
I don't get this

OK it will exit, but then what's the purpose of having an error in the signature of `lint` method

Reply:
@ccoVeille good point, if last step in lint method is os.Exit(1) so error in the signature is useless :)

Reply:
I would have kept the signature and refactored to return an error

I would recommend using this

https://pkg.go.dev/golang.org/x/sync/errgroup#Go

https://pkg.go.dev/golang.org/x/sync/errgroup#Wait

It's what you need